### ISSUE TITLE
Brunch doesn't compile files at all since version 2.7.2, if they aren't located in the default directories ('app', 'test', 'vendor')

### ISSUE THREAD

**Issue by r-zzi** [brunch/brunch#1313]
### Description

I'm continuing learning Elixir/Phoenix using the default `mix phoenix.new` app, and yesterday I got an issue after receiving the new Brunch version 2.7.2. It just does not compile anything at all.

I looked at the changes in the last version and debugged the Brunch code a little bit. 
I found out that if I comment line 438 in `config.js`, Brunch works correctly, as the previous version.

```javascript
paths.watched = ['app', 'test', 'vendor'].map(joinRoot);
```

It seems that this line overwrites the value specified in the config with the default value (`'app', 'test', 'vendor'`) and Brunch doesn't watch my files at all (they are located in `web/static`, my config is attached below, it's the default config of the `phoenix.new` app).

### Expected behavior

I run `brunch -build` and it compiles .js files into `app.js` as described in the config file.

### Actual behavior

`app.js` isn't created at all

### Environment

1. Brunch: 2.7.2
2. Node: 5.10.1
3. NPM: 3.8.3
4. Operating system: Ubuntu 15.10

### `package.json` contents

```json
{
  "repository": {
  },
  "dependencies": {
    "babel-brunch": ">=6.0.0",
    "brunch": "2.7.2",
    "clean-css-brunch": ">=1.8.0",
    "css-brunch": ">=1.7.0",
    "javascript-brunch": ">=1.8.0",
    "uglify-js-brunch": ">=1.7.0",
    "phoenix": "file:deps/phoenix",
    "phoenix_html": "file:deps/phoenix_html"
  }
}
```

### brunch config contents

```javascript
  files: {
    javascripts: {
      joinTo: "js/app.js"
    },
    stylesheets: {
      joinTo: "css/app.css"
    },
    templates: {
      joinTo: "js/app.js"
    }
  },

  conventions: {
    assets: /^(web\/static\/assets)/
  },

  paths: {
    watched: [
      "web/static",
      "test/static"
    ],
    public: "priv/static"
  },

  plugins: {
    babel: {
      ignore: [/web\/static\/vendor/]
    }
  },

  modules: {
    autoRequire: {
      "js/app.js": ["web/static/js/app"]
    }
  },

  npm: {
    enabled: true,
    whitelist: ["phoenix", "phoenix_html"]
  }
};
```

### Other useful files, when present (log, `bower.json` etc.)

```
@ubuntu:~/Sources/Elixir/sketch-guess$ ./node_modules/brunch/bin/brunch --version
2.7.2
@ubuntu:~/Sources/Elixir/sketch-guess$ ./node_modules/brunch/bin/brunch build -d
  brunch:config Trying to load brunch-config +0ms
  brunch:plugins Loaded plugins: babel-brunch, css-brunch, javascript-brunch +375ms
  brunch:watch add package.json +19ms
  brunch:watch add brunch-config.js +1ms
  brunch:watch add test/test_helper.exs +6ms
  brunch:list Reading test/test_helper.exs +1ms
  brunch:watch add test/controllers/page_controller_test.exs +10ms
  brunch:list Reading test/controllers/page_controller_test.exs +1ms
  brunch:watch add test/support/channel_case.ex +0ms
  brunch:list Reading test/support/channel_case.ex +0ms
  brunch:watch add test/support/conn_case.ex +0ms
  brunch:list Reading test/support/conn_case.ex +0ms
  brunch:watch add test/support/model_case.ex +1ms
  brunch:list Reading test/support/model_case.ex +0ms
  brunch:watch add test/views/error_view_test.exs +0ms
  brunch:list Reading test/views/error_view_test.exs +1ms
  brunch:watch add test/views/layout_view_test.exs +1ms
  brunch:list Reading test/views/layout_view_test.exs +0ms
  brunch:watch add test/views/page_view_test.exs +0ms
  brunch:list Reading test/views/page_view_test.exs +0ms
  brunch:write Writing 0/0 files +69ms
24 Apr 08:22:01 - info: compiled in 757ms

```

**Comment by goshakkk** [comment]
Wow, thanks for pointing that out! I must have forgotten to change it to `paths.watched.map`... https://github.com/brunch/brunch/pull/1314 will fix it! 🙈

Sorry for that!

**Comment by r-zzi** [comment]
Thanks! :)

I'd like to use an opportunity and ask a kind of off-topic question about practices in Brunch.
Let's imagine I made some changes / features / refactorings and want to create a pull request.
1. Should I write some tests for my changes or new features?
2. How could I make sure that I didn't break something?

Brunch isn't a very small project and it's suggested as a 'default solution' for some frameworks, but I found only ~10 tests in this repository. Is it something like an issue or there are some other ways to ensure its consistency? :)

Thank you again!
